src/components/EditAccountNameDialog.js

```diff
@@ -0,0 +1,45 @@
+import React, {useState} from "react";
+import DialogForm from "./DialogForm";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import TextField from "@material-ui/core/TextField";
+import Button from "@material-ui/core/Button";
+import DialogActions from "@material-ui/core/DialogActions";
+
+export default function EditAccountNameDialog({open, oldName, onClose, onEdit}) {
```

Comment:
Instead of a dialog, you could make the name field directly editable. I don't have strong feels though.